src/TaskViewSection/Toolbar/ToolbarButtons.jsx

```diff
@@ -1,159 +1,158 @@
+                removeFromList()
+                showAlert({
+                    title: 'All done tasks has been removed successfully',
+                    type: 'success',
+                    background: 'bg-success',
+                    isShowed: true,
+                })
+            } else {
+                showAlert({
+                    title: 'There are no completed tasks to be deleted',
+                    type: 'error',
+                    background: 'bg-error',
+                    isShowed: true,
+                })
+            }
+        }
     }
-    return false;
-  };
 
-  const showAlert = (params) => {
-    setAlert({
-      title: params.title,
-      type: params.type,
-      background: params.background,
-      isShowed: params.isShowed,
-    });
-    setTimeout(() => {
-      setAlert({ isShowed: false });
-    }, 3000);
-  };
-  const removeTasksHandler = () => {
-    isAnyTaskDone()
-      ? showAlert({
-          title: "All done tasks has been removed successfully",
-          type: "success",
-          background: "bg-success",
-          isShowed: true,
+    const showAlert = (alertData) => {
+        setAlert({
+            title: alertData.title,
+            type: alertData.type,
+            background: alertData.background,
+            isShowed: alertData.isShowed,
+        })
+        setTimeout(() => {
+            setAlert({ isShowed: false })
+        }, 3000)
+    }
+    const makeAllTasksDone = () => {
+        const newList = [...taskList]
+        taskList.forEach((task) => {
+            if (
+                task.category === selectedTabCategory ||
+                selectedTabCategory === 'all'
+            ) {
+                task.done = true
+            }
+        })
+        setTaskList(newList)
+    }
+    const undoneAllTasks = () => {
+        const newList = [...taskList]
+        newList.forEach((task) => {
+            if (
+                task.category === selectedTabCategory ||
+                selectedTabCategory === 'all'
+            ) {
+                task.done = false
+            }
         })
-      : showAlert({
-          title: "There are no completed tasks to be deleted",
-          type: "error",
-          background: "bg-error",
-          isShowed: true,
-        });
-  };
-  const makeAllTasksDone = () => {
-    const newList = [...taskList];
-    newList.forEach((task) => {
-      if (
-        task.category === selectedTabCategory ||
-        selectedTabCategory === "all"
-      ) {
-        task.done = true;
-      }
-    });
-    setTaskList(newList);
-  };
-  const undoneAllTasks = () => {
-    const newList = [...taskList];
-    newList.forEach((task) => {
-      if (
-        task.category === selectedTabCategory ||
-        selectedTabCategory === "all"
-      ) {
-        task.done = false;
-      }
-    });
-    setTaskList(newList);
-  };
-  function RemoveDoneTasksButton({ action }) {
+        setTaskList(newList)
```

Comment:
let's think how to rewrite this without mutating task state, I mean `task.done = false` or `task.done = true`?
small tip take a look on `.map` function